switchlink/switchlink_main.c

```diff
@@ -321,12 +339,16 @@ void *switchlink_main(void *args) {
   return NULL;
 }
 
-int switchlink_stop(void) {
+void *switchlink_stop(void *args) {
+  pthread_mutex_lock(&rpc_stop_lock);
+  while(!rpc_stop_cookie) {
```

Comment:
You might want to add a space between `while` and `(`. It's a control statement, not a function call.